scipy/_lib/_disjoint_set.py

```diff
@@ -210,6 +210,25 @@ def subset(self, x):
             nxt = self._nbrs[nxt]
         return set(result)
 
+    def subset_size(self, x):
+        """Get the size of the subset containing `x`.
+
+        Note that this method is faster than ``len(self.subset(x))`` because
+        the size is directly read off an internal field, without the need to
+        instantiate the full subset.
+
+        Parameters
+        ----------
+        x : hashable object
+            Input element.
+
+        Returns
+        -------
+        result : int
+            Size of the subset containing `x`.
+        """
+        return self._sizes[self[x]]
```

Comment:
Other methods here are diligent about checking performing a

```python
        if x not in self._indices:
            raise KeyError(x)
```

before doing anything. Probably not a must, as the same `KeyError` will surface almost immediately anyway.

Reply:
Could be a good thing to have a better error message. But to follow our conventions it should be `raise ValueError(message)`. And this would need a test with `pytest.raises(..., match=message)`

Reply:
Which does not necessarily mean it's perfect 😉 I will let @fuglede make the call.

Reply:
Good point, sounds good to me.